src/lib/AttestationLib.sol

```diff
@@ -6,15 +6,34 @@ import { SSTORE2 } from "solady/utils/SSTORE2.sol";
 
 library AttestationLib {
     // The hash of the data type used to relay calls to the attest function. It's the value of
-    bytes32 internal constant ATTEST_TYPEHASH = keccak256("AttestationRequest(address,uint48,bytes,uint256[])");
+    bytes32 internal constant ATTEST_REQUEST_TYPEHASH = keccak256("AttestationRequest(address,uint48,bytes,uint256[])");
+    bytes32 internal constant ATTEST_TYPEHASH =
+        keccak256("SignedAttestationRequest(AttestationRequest,uint256 nonce)AttestationRequest(address,uint48,bytes,uint256[])");
```

Comment:
is this correct? ie do you just add the two strings next to each other?

Reply:
https://eips.ethereum.org/EIPS/eip-712#definition-of-encodetype

> If the struct type references other struct types (and these in turn reference even more struct types), then the set of referenced struct types is collected, sorted by name and appended to the encoding. An example encoding is `Transaction(Person from,Person to,Asset tx)Asset(address token,uint256 amount)Person(address wallet,string name).`